### PR TITLE
Fix errors on PWM output

### PR DESCRIPTION
**Pull request by pkendall64**
There were a couple of problems detected when testing 16 channel output...

The `_activePwmChannels` variable was being used for channels and pins, so it's been change to use channel only.
This also resolved the problem were there was an overflow when using GPIO pins numbered 32 or greater.

The `allocateLedcChn` function has been rewritten so that a LEDC timer can be used if a channel (PWM pin) within the same group (high/low speed) is set to the same frequency. This allows up to 4 different frequency based PWM or Duty-cycle within the same group to be used simultaneously.

Previously the pins were effectively pair-wise, so each channel pair had to be using the same frequency.

### CODE REVIEW

**Comment by JyeSmith** [comment]
Looks good!  Below is the config used for testing,

It can be 'broken' with really weird setups and lots of various modes (timers allocations). But I think this is going to be fine for 99.999999% of users.

image

**Review by JyeSmith — APPROVED** [review]

File: src/lib/ServoOutput/ServoMgr.cpp [hunk 1]
```diff
@@ -15,38 +15,81 @@ ServoMgr::ServoMgr(const uint8_t *const pins, const uint8_t outputCnt, uint32_t
...
         }
-        else if (_timerConfigs[i].freq == target_freq && _timerConfigs[i].ch2 == 255)
+        if (_timerConfigs[timer_idx] == target_freq)
         {
-            _timerConfigs[i].ch2 = ch;
-            ledc_chn = i * 2 + 1;
-            break;
+            ledcAttachPinEx(pin, ch, (ledc_timer_t)i);
+            DBGLN("allocate ledc_ch %d on pin %d using group: %d, ledc_tim: %d, bits: %d", ch, pin, group, i, _resolution_bits[ch]);
+            return;
```
Comment (Ncerzzk) [inline src/lib/ServoOutput/ServoMgr.cpp:89]: here may be problems when 3 channels using the same freq?
like:
ch0 50hz
ch1 50hz
ch2 50hz

Comment (pkendall64) [reply src/lib/ServoOutput/ServoMgr.cpp:89]: No thats fine. All the channels in a group can share the same timer!

**Comment by Ncerzzk** [comment]
late to see this pr. Nicer implement!  I will do some test when I go home ^ ^

File: src/lib/ServoOutput/ServoMgr.cpp (hunk 1 above)
Comment (Ncerzzk) [reply src/lib/ServoOutput/ServoMgr.cpp:89]: > No thats fine. All the channels in a group can share the same timer!
Checked in the datasheet. You are right!

**Review by Ncerzzk — APPROVED** [review]

**Review by wvarty — APPROVED** [review]

**Pull request merged**